### ISSUE TITLE
Invalid User-Agent

### ISSUE THREAD

**Issue by xu-cheng** [gjtorikian/html-proofer#634]
Currently, the user-agent is passed as part of HTTP headers in `Typhoeus.get(url, options)`. However, this actually results to incorrect user-agent being emitted. You can observe this by passing `--typhoeus-config={"verbose":true}`.

```
[snip]
Get / HTTP/2
Host: linkedin.com
User-Agent: Typhoeus - https://github.com/typhoeus/typhoeus
User-Agent: Mozilla/5.0 (Macintosh; Intel Mac OS X 10_15_6) AppleWebKit/605.1.15 (KHTML, like Gecko) Version/14.0 Safari/605.1.15
Accept: application/xml,application/xhtml+xml,text/html;q=0.9, text/plain;q=0.8,image/png,*/*;q=0.5
[snip]
```

Noted that there are two user-agent. This could cause certain website reporting 404 error.

It seems that you should use `Typhoeus::Config.user_agent = "custom user agent"` to set the user-agent per [the document](https://rubydoc.info/github/typhoeus/typhoeus/Typhoeus%2FConfig:user_agent).

**Comment by gjtorikian** [comment]
Hmm. Could you show me the exact command you are running? For example, `htmlproofer --as-links https://linkedin.com --typhoeus-config='{"verbose":true}'` does show only one UA for me.

**Comment by xu-cheng** [comment]
This is the command I used:
```
 htmlproofer --typhoeus-config='{"verbose":true,"headers":{"User-Agent":"Mozilla/5.0 (Macintosh; Intel Mac OS X 10_15_6) AppleWebKit/605.1.15 (KHTML, like Gecko) Version/14.0 Safari/605.1.15"}}' --as-links https://linkedin.com
```

Here are the log:
```
$ bundle exec htmlproofer --typhoeus-config='{"verbose":true,"headers":{"User-Agent":"Mozilla/5.0 (Macintosh; Intel Mac OS X 10_15_6) AppleWebKit/605.1.15 (KHTML, like Gecko) Version/14.0 Safari/605.1.15"}}' --as-links https://linkedin.com
Running ["LinkCheck"] on ["https://linkedin.com"]...


Checking 1 external link...
  Trying 13.107.42.14...
TCP_NODELAY set
Connected to linkedin.com (13.107.42.14) port 443 (#0)
ALPN, offering h2
ALPN, offering http/1.1
successfully set certificate verify locations:
  CAfile: /etc/ssl/cert.pem
  CApath: none
TLSv1.2 (OUT), TLS handshake, Client hello (1):
[snip]
Server certificate:
 subject: C=US; ST=California; L=Sunnyvale; O=LinkedIn Corporation; CN=www.linkedin.com
 start date: Feb 17 00:00:00 2021 GMT
 expire date: Aug 16 23:59:59 2021 GMT
 subjectAltName: host "linkedin.com" matched cert's "linkedin.com"
 issuer: C=US; O=DigiCert Inc; CN=DigiCert SHA2 Secure Server CA
 SSL certificate verify ok.
Using HTTP2, server supports multi-use
Connection state changed (HTTP/2 confirmed)
Copying HTTP/2 data in stream buffer to connection buffer after upgrade: len=0
Using Stream ID: 1 (easy handle 0x7ffb20be0600)
HEAD / HTTP/2
Host: linkedin.com
Accept: */*
User-Agent: Typhoeus - https://github.com/typhoeus/typhoeus
User-Agent: Mozilla/5.0 (Macintosh; Intel Mac OS X 10_15_6) AppleWebKit/605.1.15 (KHTML, like Gecko) Version/14.0 Safari/605.1.15

Connection state changed (MAX_CONCURRENT_STREAMS == 100)!
HTTP/2 400
content-length: 339
content-type: text/html; charset=us-ascii
date: Sat, 17 Apr 2021 05:08:26 GMT

Connection #0 to host linkedin.com left intact
Found bundle for host linkedin.com: 0x7ffb228b6b10 [can multiplex]
Re-using existing connection! (#0) with host linkedin.com
Connected to linkedin.com (13.107.42.14) port 443 (#0)
Using Stream ID: 3 (easy handle 0x7ffb20be0600)
GET / HTTP/2
Host: linkedin.com
Accept: */*
User-Agent: Typhoeus - https://github.com/typhoeus/typhoeus
User-Agent: Mozilla/5.0 (Macintosh; Intel Mac OS X 10_15_6) AppleWebKit/605.1.15 (KHTML, like Gecko) Version/14.0 Safari/605.1.15

HTTP/2 400
content-type: text/html; charset=us-ascii
date: Sat, 17 Apr 2021 05:08:26 GMT
content-length: 339

Connection #0 to host linkedin.com left intact
-
  *  External link https://linkedin.com failed: 400 No error

HTML-Proofer found 1 failure!
* Closing connection 0
```

So the issue seems come from that I set my own User-Agent.

**Comment by gjtorikian** [comment]
Good find, this was a very strange bug but it should be fixed in 3.19.1.
